lib/fluent/plugin/in_tail/position_file.rb

```diff
@@ -248,6 +248,20 @@ def read_inode
       end
     end
 
-    TargetInfo = Struct.new(:path, :ino)
+    TargetInfo = Struct.new(:path, :ino) do
+      def ==(other)
+        return false unless other.is_a?(TargetInfo)
+        self.path == other.path
+      end
+
+      def hash
+        self.path.hash
```

Comment:
I'm not sure follow_inodes feature but does it work well also on `@follow_inodes`?
I think that keys of `@tails` should be inode on `@follow_inodes`.

Reply:
Or paths are updated by `update_watcher` on rotation so paths are never duplicate?

Reply:
I'll take a look on this.

Reply:
I'm checking it too.
Probably above my concern is wrong, it should be path maybe.
Otherwise `update_watcher` can't discard old `TailWather` from `@tails`.

Reply:
Memo:

Instead of overriding them, I thought using `target_info.path` as a key is more straightforward and understandable.
`@tails[target_info.path]`

But it's also wrong.
`TargetInfo` is referred by `each_keys` so keys have to `TargetInfo`.

Reply:
Hmm, probably it's O.K. as a first aid but I think more refactoring is needed to make easy to understand in the future.

Reply:
> Instead of overriding them, I thought using `target_info.path` as a key is more straightforward and understandable: `@tails[target_info.path]`
> 
> But it's also wrong.
> `TargetInfo` is referred by `each_keys` so keys have to be `TargetInfo`.

Yes! I'm also considering to use it to be made easier to understand, but overriding equality on TargetInfo is one of the simplest way to dispose discarded old TailWatcher. (This strategy dosen't request more refactoring.)